app/src/androidTest/java/com/github/mateo762/myapplication/profile/ProfileViewModelTest.kt

```diff
@@ -0,0 +1,164 @@
+package com.github.mateo762.myapplication.profile
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.github.mateo762.myapplication.followers.UserRepository
+import com.google.firebase.auth.FirebaseAuth
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.resetMain
+
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.After
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.jupiter.api.assertDoesNotThrow
+import org.mockito.Mockito.mock
+
+@ExperimentalCoroutinesApi
+@HiltAndroidTest
+class ProfileViewModelTest {
+
+    @get:Rule
+    val hiltRule = HiltAndroidRule(this)
+
+    @get:Rule
+    val instantExecutorRule = InstantTaskExecutorRule()
```

Comment:
What is this rule needed for?

Reply:
It is needed for testing coroutines with runTest block

Reply:
Okay.